pymc_bart/pgbart.py

```diff
@@ -456,17 +465,23 @@ def get_split_value(available_splitting_values, idx_data_points, missing_data):
 
 
 @njit
-def draw_leaf_value(y_mu_pred, m, norm, shape):
+def draw_leaf_value(y_mu_pred, x_mu, m, norm, shape, response):
     """Draw Gaussian distributed leaf values."""
+    linear_params = None
+    mu_mean = np.empty(shape)
     if y_mu_pred.size == 0:
-        return np.zeros(shape)
+        return np.zeros(shape), linear_params
 
     if y_mu_pred.size == 1:
         mu_mean = np.full(shape, y_mu_pred.item() / m)
     else:
-        mu_mean = fast_mean(y_mu_pred) / m
-
-    return norm + mu_mean
+        if response == "linear":
+            y_fit, linear_params = fast_linear_fit(x=x_mu, y=y_mu_pred)
+            mu_mean = fast_mean(y_fit.reshape(1, -1)) / m
```

Comment:
@aloctavodia Now I remember why I though about reducing with the mean: because otherwise numba does not compile because a problems in dimensions (see ci)
```python
Cannot unify array(float64, 2d, C) and array(float64, 1d, C) for 'mu_mean.4', defined at /home/runner/work/pymc-bart/pymc-bart/pymc_bart/pgbart.py (483)
```
This is what has been driving me crazy 😅

Reply:
For the moment we can just comment out `@njitt` for that function. Not sure how much numba is helping with that function anyways. But if we still want numba we may need to refactor `draw_leaf_values` into one or more jitted functions with a well-defined output and one unfitted function with a variable output.

Reply:
In https://github.com/pymc-devs/pymc-bart/pull/79/commits/996596f2053bd4bb1b942a5b483d712d84b042c0 I changed the linear fir formula to 
```python
b = ((x - xbar) @ (y - ybar).T) / ((x - xbar) @ (x - xbar).T)
```
and `numba` stop complaining :)

Reply:
solved in https://github.com/pymc-devs/pymc-bart/pull/79/commits/f2c546a33059fd8f96375f5fb8a81b2f364a7fc0